### PR TITLE
add autogenerate slug function

### PR DESCRIPTION
**Pull request by unabomber87**
add URL slug generate function when user enter a title for a post

### CODE REVIEW

File: publishable/views/posts/edit-add.blade.php [hunk 1]
```diff
@@ -209,4 +209,13 @@
 @section('javascript')
     <script src="{{ config('voyager.assets_path') }}/lib/js/tinymce/tinymce.min.js"></script>
     <script src="{{ config('voyager.assets_path') }}/js/voyager_tinymce.js"></script>
+    <script src="{{ config('voyager.assets_path') }}/js/speakingurl.js"></script>
+    <script type="text/javascript">
+        $(document).ready(function() {
+            $('[name="title"]').on('keyup', function() {
+                //console.log( $(this).val() );
```
Comment (marktopper) [inline publishable/views/posts/edit-add.blade.php]: Please remove `//console.log( $(this).val() );`

**Comment by marktopper** [comment]
Shouldn't this be for other BREAD 🍞 as well? Now it will only be for posts, but how about `categories` and when I use Voyager to create a `products` page?

**Comment by unabomber87** [comment]
i change it to be a global function so
when there's an input with name 'title' and another one with name 'slug' => generate slug in input slug
or an input with name 'name' and another one with name 'slug' => generate slug in input slug

**Review by akazorg — COMMENTED** [review]
Will it update the slug input even if it isn't empty?
It would be more flexible if you can define on Bread Option, what field is causing the auto generation.

**Comment by marktopper** [comment]
@akazorg: You are right, this should become a bread option.

**Comment by akazorg** [comment]
@marktopper I wrote a slugify plugin in JS for working with the BREAD, it works great.

Basically you can define:
* what field is causing the update
* if you want to force slug generation, for non empty slug fields.

By default, if slug is blank it will auto generate a new one.

I can share it if you want.

**Comment by marktopper** [comment]
Please share @akazorg, might be a better approach.

**Comment by akazorg** [comment]
@marktopper I've found a little nasty bug, will fix it and share with you!

**Comment by marktopper** [comment]
@akazorg: On Slack or where?

**Comment by akazorg** [comment]
I'll let you know @marktopper, sorry, I'm just so busy atm.
I hope to find some time today.

**Comment by marktopper** [comment]
It's okay @akazorg

**Pull request merged**